core/src/vote_simulator.rs

```diff
@@ -377,3 +377,879 @@ pub fn initialize_state(
         HeaviestSubtreeForkChoice::new_from_bank_forks(bank_forks.clone());
     (bank_forks, progress, heaviest_subtree_fork_choice)
 }
+
+#[cfg(test)]
+mod test {
```

Comment:
`VoteSimulator` is a testing utility. perhaps it would be easier to move it somewhere else instead of copying all the tests into here?
I think the tests should stay close to the impls they are testing.